cli/cli.go

```diff
@@ -62,13 +62,16 @@ var RootCmd = &cobra.Command{
 		go func(ch chan struct{}) {
 			for {
 				mu.Lock()
-				<-ch
+				_, isOpen := <-ch
+				if !isOpen {
+					mu.Unlock()
+					return
+				}
```

Comment:
I think the same logic is needed above on lines 47-49

I find it a little odd that a channel is returned on line 40 to be passed to `m` on lines 51 and 74.

I _think_ what this should do is return a _read only_ channel (`<- chan struct{}` instead of just `chan struct{}`) from mtr.NewMTR, and _store_ the channel in the structure, and then close it before `Run` returns. Since `Run` would have access to the channel, it wouldn't need it as an argument.